### PR TITLE
Event Enabling/Disabling features

### PR DESCRIPTION
**Pull request by veritas4**
There's quite a lot of changes in this PR. Here are the things I did:

- Added "event_enabled" and "officer_in_charge" columns to Events
- Deleted "date" and "time" columns, and added the "datetime" column to Events
- Added a checkbox in the Events form to enable/disable signup for an event
- Implemented logic to disable signup for an event if it already happened.
- Fixed existing test cases that were affected by changing the attributes of the event.
- Deleted the officer controllers, still need to delete the officer views.
- Wrote test cases for this new feature

### CODE REVIEW

File: app/controllers/admin/events_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 module Admin
   class EventsController < ApplicationController
-    before_action :set_event, only: %i[show edit update destroy]
+    before_action :set_event, :update_event_status, only: %i[show edit update destroy]
```
Comment (jrr2024) [inline app/controllers/admin/events_controller.rb:5]: Few things:

1. We don't want to do all this work any time these routes are hit. Instead, use ActiveJob to schedule a disabling of the event on its datetime.
2. Even if we were to keep it this way, we'd need to perform this on ALL event actions including index for all types of users.

I'd just make a collection instance variable @active_event_logs in the index action that queries for all event logs that aren't disabled.

**Review by jrr2024 — CHANGES_REQUESTED** [review]

File: app/controllers/admin/events_controller.rb (hunk 1 above)
Comment (veritas4) [reply app/controllers/admin/events_controller.rb]: Due to the complexity of implementing job scheduling, we chose an alternative way to implement this without having to do N+1 work for every route.

**Review by veritas4 — COMMENTED** [review]
Add function in Event model to check if the event is active. Use in Views to stop users from signing up for an event that is expired.

File: app/controllers/admin/events_controller.rb [hunk 2]
```diff
@@ -12,6 +12,7 @@ def index
     # GET /events/1 or /events/1.json
     def show
       @event = Event.find(params[:id])
+      @event.update_attribute(:event_enabled, @event.is_active?)
```
Comment (jrr2024) [inline app/controllers/admin/events_controller.rb]: no need

**Review by jrr2024 — APPROVED** [review]

**Pull request merged**